### ISSUE TITLE
Remove randomness from test

### ISSUE THREAD

**Issue by kellertuer** [JuliaManifolds/Manopt.jl#101]
Currently, some tests are using random (despite those testing random generation). This should be removed to deterministically hit all lines.

**Comment by mateuszbaran** [comment]
Getting random numbers for tests should be fine if you just set the seed to a constant, right? That's what we do in Manifolds.jl.

**Comment by kellertuer** [comment]
Yeah, but currently a whole Quasi Newton run for example is based on random numbers (not s test I wrote). I want to make that deterministic.

**Comment by mateuszbaran** [comment]
`rand()` actually *is* deterministic :wink: .

**Comment by mateuszbaran** [comment]
At least for a single version of Julia.

**Comment by kellertuer** [comment]
...and I just upped the tests from 1.6 to 1.7 where they changed the default seed.

**Comment by mateuszbaran** [comment]
Yes, so the solution is to either hardcode examples or use https://github.com/JuliaRandom/StableRNGs.jl .

**Comment by kellertuer** [comment]
Maybe hardcoded is nicer, but let'S for now just keep track here to do that somewhen.

**Comment by kellertuer** [comment]
Update: The main part to redo is the Quasi newton tests – I will look into these next.
